tests: drivers: Added overlays to tests for 063_ble

pavloinfineon opened the pull request:
Update overlay for adc
Added overlay for gpio_basic_api
Added overlay for i2c_api
Updated board.cmake

MaureenHelm commented on boards/arm/cy8cproto_063_ble/board.cmake:
```diff
@@ -5,3 +5,6 @@
 
 include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
+
+board_runner_args(pyocd "--target=cy8c6xx7_nosmif")
+include(${ZEPHYR_BASE}/boards/common/pyocd.board.cmake)
```
These changes are unrelated to overlays and need to go in a separate commit.

pavloinfineon replied:
Split into 2 commits

MaureenHelm reviewed (changes requested):
```
tests: drivers: Added overlays to tests for 063_ble

Update overlays for adc
Added overlay for gpio_basic_api
Added overlay for i2c_api
Updated board.cmake

Signed-off-by: Pavlo Havrylyuk <pavlo.havrylyuk@infineon.com>
```

Please explain in the commit message **why** you're making these changes

pavloinfineon committed:
boards: cy8cproto_063_ble: Add pyocd support

Allow support for flashing board with pyocd

Signed-off-by: Pavlo Havrylyuk <pavlo.havrylyuk@infineon.com>

pavloinfineon committed:
tests: drivers: Added overlays to tests for 063_ble

Fixed pinout in adc overlays to align with board
Allowed adc driver sample
Added overlay for gpio_basic_api test support
Added overlay for i2c_api test support

Signed-off-by: Pavlo Havrylyuk <pavlo.havrylyuk@infineon.com>
